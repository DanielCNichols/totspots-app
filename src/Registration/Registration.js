import React, { useState } from 'react';
import CustomFormHook from '../Hooks/customHooks';
import AuthService from '../services/AuthService';
import { ReactComponent as RegistrationPic } from '../assets/family_.svg';
import s from './Registration.module.css';

//Let's simplify this and use just the F,L, EMail, passowrd, and zip
export default function RegistrationForm() {
  let [error, setError] = useState(null);
  const {
    inputs,
    handleInputChange,
    handleSubmit,
    touched,
    handleTouched,
  } = CustomFormHook(async () => {
    try {
      let user = {
        email: inputs.email,
        password: inputs.password,
        first_name: inputs.first_name,
        last_name: inputs.last_name,
      };
      let res = await AuthService.postUser(user);
    } catch (err) {
      setError(err);
    }
  });

  return (
    <div className={s.wrapper}>
      <form className={s.registration} onSubmit={handleSubmit} noValidate>
        <legend>Sign Up</legend>
        <div className={s.formElement}>
          <input
            className={s.regInput}
            styles={{ position: 'absolute' }}
            type="text"
            id="first_name"
            name="first_name"
            onBlur={handleTouched}
            onChange={handleInputChange}
            required
          />
          <label className={s.regLabel} htmlFor="first_name">
            First Name
          </label>
        </div>
        <div className={s.validationError}>
          {touched.first_name ? <p>First name is required</p> : null}
        </div>
        <div className={s.formElement}>
          <input
            className={s.regInput}
            type="text"
            id="last_name"
            name="last_name"
            onBlur={handleTouched}
            onChange={handleInputChange}
            required
          />
          <label className={s.regLabel} htmlFor="last_name">
            Last Name
          </label>
        </div>
        <div className={s.validationError}>
          {touched.last_name ? <p>Last name is required</p> : null}
        </div>
        <div className={s.formElement}>
          <input
            className={s.regInput}
            onBlur={handleTouched}
            onChange={handleInputChange}
            name="email"
            type="text"
            id="email"
            required
          />
          <label className={s.regLabel} htmlFor="email">
            Email
          </label>
          {/* here we can put some custom validation */}
        </div>
        <div className={s.validationError}>
          {touched.email ? <p>Email is required</p> : null}
        </div>
        <div className={s.formElement}>
          <input
            className={s.regInput}
            type="text"
            id="password"
            name="password"
            onBlur={handleTouched}
            onChange={handleInputChange}
            required
          />
          <label className={s.regLabel} htmlFor="password">
            Password
          </label>
        </div>
        <div className={s.validationError}>
          {touched.password ? <p>Password is required</p> : null}
        </div>
        <button>Sign Up</button>
        {error ? <p>{error.error}</p> : null}
      </form>
      <div className={s.imageContainer}>
        <div>
          <h3>Almost There!</h3>
          You are close to getting some things asbout stuff for restaurants. Get
          money, take money.
        </div>
        <RegistrationPic />
      </div>
    </div>
  );
}
