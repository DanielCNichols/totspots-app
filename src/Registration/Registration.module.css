* {
  box-sizing: border-box;
}

.registration {
  position: relative;
  top: 100px;
  margin: 0 auto;
  width: 100%;
  padding: 0 10px;
}

.formElement {
  position: relative;
  overflow: hidden;
  margin: 20px auto;
}

.regLabel {
  transition: background 0.2s, color 0.2s, top 0.2s, bottom 0.2s right 0.2s,
    left 0.2s;
  position: absolute;
  color: #999;
  padding: 7px 6px;
  top: 0;
  bottom: 0;
  left: 0;
  width: 100%;
  text-align: left;
}

.regInput {
  width: 100%;
  border: 2px solid gray;
  background: none;
  position: relative;
  top: 0;
  left: 0;
  z-index: 1;
  padding: 12px;
  outline: 0;
  font-size: 1em;
}

.regInput:valid {
  background: white;
}

.regInput:focus {
  border-color: lightblue;
  padding: 3px 6px 20px 6px;
}

.regInput:focus + label {
  background: lightblue;
  color: black;
  font-size: 90%;
  padding: 1px 6px;
  z-index: 2;
  text-transform: uppercase;
  top: 90%;
  margin-top: -16px;
}

.validationError {
  color: red;
}

@media screen and (min-width: 450px) {
  .registration {
    width: 320px;
    top: 200px;
  }
}

@media screen and (min-width: 800px) {
  .wrapper {
    width: 80%;
    display: flex;
    justify-content: space-around;
    position: relative;
    top: 200px;
    margin: 0 auto;
  }

  .regInput:focus + label {
    background: lightblue;
    color: black;
    font-size: 70%;
    padding: 1px 6px;
    z-index: 2;
    text-transform: uppercase;
    top: 100%;
    margin-top: -16px;
  }

  .registration {
    position: initial;
  }

  .imageContainer {
    width: 50%;
  }

  .imageContainer > svg {
    width: 100%;
  }
}
