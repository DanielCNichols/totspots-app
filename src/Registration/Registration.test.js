import React from 'react';
import ReactDOM from 'react-dom'
import Registration from './Registration.js'
import renderer from 'react-test-renderer'


it('Registration renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Registration />, div);
  ReactDOM.unmountComponentAtNode(div)
})

it('renders Registration as expected', () => {
  const tree = renderer
  .create(
    <Registration name="Registration"/>
  )
  .toJSON();
  expect(tree).toMatchSnapshot();
})
