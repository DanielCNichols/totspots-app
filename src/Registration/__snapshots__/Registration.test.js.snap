// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Registration as expected 1`] = `
<section>
  <header>
    <h2>
      Create New Account
    </h2>
  </header>
  <form
    action=""
  >
    <fieldset>
      <label
        htmlFor="Name"
      >
        Name
        <input
          placeholder="Jon Doe"
          required={true}
          type="text"
        />
      </label>
      <label
        htmlFor="email"
      >
        Email
        <input
          id="email"
          name="email"
          placeholder="jon.doe@mindspring.com"
          type="text"
        />
      </label>
      <label
        htmlFor="City"
      >
        City
        <input
          placeholder="Durham"
          type="text"
        />
      </label>
      <label
        htmlFor="state"
      >
        State
        <input
          id="state"
          max-length="2"
          name="state"
          placeholder="NC"
          required={true}
          type="text"
        />
      </label>
      <label
        htmlFor="UserName"
      >
        User Name
        <input
          placeholder="jdoe1234!"
          required={true}
          type="text"
        />
      </label>
      <label
        htmlFor="password"
      >
        Password
        <input
          id="password"
          name="password"
          type="text"
        />
      </label>
      <label
        htmlFor="confirm"
      >
        Confirm Password
        <input
          id="confirm"
          name="confirm"
          type="text"
        />
      </label>
    </fieldset>
    <button>
      Submit
    </button>
    <button>
      Cancel
    </button>
  </form>
</section>
`;
